c10/macros/Macros.h

```diff
@@ -361,22 +363,32 @@ extern SYCL_EXTERNAL void __assert_fail(
     const char* func);
 #else // __SYCL_DEVICE_ONLY__
 #if (defined(__CUDA_ARCH__) && !(defined(__clang__) && defined(__CUDA__)))
+// CUDA supports __assert_fail function which are common for both device
+// and host side code.
 __host__ __device__
-#endif // __CUDA_ARCH__
+#endif
+
+    // This forward declaration matching the declaration of __assert_fail
+    // exactly how it is in glibc in case parts of the program are compiled with
+    // different NDEBUG settings. Otherwise we might get 'ambiguous declaration'
+    // error. Note: On ROCm - this declaration serves for host side compilation.
     void
     __assert_fail(
         const char* assertion,
         const char* file,
         unsigned int line,
-        const char* function) throw()
-// We match the declaration of __assert_fail exactly how it is in glibc in case
-// parts of the program are compiled with different NDEBUG settings. Otherwise
-// we might get 'ambiguous declaration' error.
-#ifdef __GNUC__
-        __attribute__((__noreturn__))
-#endif
```

Comment:
Can you provide a short explanation, why are you dropping this conditional? (I guess the answer is: CUDA no longer supported on Mac)

Reply:
outline of final code is like
```
#if CUDA
__host__ __device__
#endif
void __assert_fail(...) throw() __attribute__((__noreturn__));

#if HIP
// A comment on a reason why it is needed
__device__ __attribute__((noinline)) __attribute__((weak)) void __assert_fail(...);
#endif
```

So essentially this `__attribute__((__noreturn__))` is kept active for CUDA case and CPU compilation code.
And build is successful is on Mac.